pglookout/pglookout.py

```diff
@@ -483,29 +485,38 @@ def consider_failover(self, own_state, master_node, standby_nodes):
                 self.current_master,
                 master_node,
             )
-            if self.current_master:
-                self.cluster_monitor_check_queue.put("Master is missing, ask for immediate state check")
-                master_known_to_be_gone = self.current_master in self.known_gone_nodes
-                now = time.monotonic()
-                config_timeout_exceeded = (now - self.cluster_nodes_change_time) >= self.missing_master_from_config_timeout
-                if master_known_to_be_gone or config_timeout_exceeded:
-                    # we've seen a master at some point in time, but now it's
-                    # not reachable or removed from configuration, perform an
-                    # immediate failover to promote one of the standbys
-                    if master_known_to_be_gone:
-                        reason = "master got removed from configuration"
-                    else:
-                        reason = "master node is not reachable"
-                    self.log.warning("Performing failover decision because %s", reason)
+            if master_node and not self._failover_on_disconnect:
```

Comment:
```suggestion
            if not self._failover_on_disconnect and master_node:
```

To only check iff the feature is enabled.